aten/src/ATen/autocast_mode.cpp

```diff
@@ -13,13 +13,11 @@ namespace at {
 namespace autocast {
 
 bool is_enabled() {
-  //return !c10::impl::tls_is_dispatch_key_excluded(DispatchKey::AutocastCUDA) ||
-  //         !c10::impl::tls_is_dispatch_key_excluded(DispatchKey::AutocastCPU);
-  return !c10::impl::tls_is_dispatch_key_excluded(DispatchKey::AutocastCUDA);
+  return !c10::impl::tls_is_dispatch_keyset_excluded(autocast_dispatch_keyset);
 }
 
 void set_enabled(bool new_enabled) {
-  //c10::impl::tls_set_dispatch_key_excluded(DispatchKey::AutocastCPU, !new_enabled);
+  c10::impl::tls_set_dispatch_key_excluded(DispatchKey::AutocastCPU, !new_enabled);
   c10::impl::tls_set_dispatch_key_excluded(DispatchKey::AutocastCUDA, !new_enabled);
```

Comment:
This looks like the consolidated `torch.autocast` semantics, didn't we agree to split in https://github.com/pytorch/pytorch/issues/55374#issuecomment-827553545 ? cc @jgong5

Reply:
The `torch.cuda.amp.autocast` and `torch.cpu.amp.autocast` are split at the python API level but consolidated at the C++ API level. Should we split even at the C++ API level as you mentioned above?

Reply:
Yes, I think they should be split at C++ level. Otherwise if you toggle CPU autocast, do you really want CUDA autocast to turn on at the same time too?

Reply:
Sure, using the separate API for CPU and CUDA.